bot/plotting/comparison_generator.py

```diff
@@ -1,275 +1,186 @@
+            params = parameter_value_generator(
+                self.parameter_values.copy(),
+                {
+                    "Current function [A]": (None, None),
+                    "Ambient temperature [K]": (265, 355),
+                },
+            )
+
+        # convert the list containing parameter values to a
+        # dictionary for pybamm.BatchStudy
+        parameter_values_for_comp = dict(
+            list(
+                enumerate([params if not self.is_experiment else self.parameter_values])
+            )
+        )
+
+        t_end = self.calculate_t_end(parameter_values_for_comp)
+
+        batch_study = pybamm.BatchStudy(
+            models=self.models_for_comp,
+            parameter_values=parameter_values_for_comp,
+            experiments=self.experiment,
+            permutations=True,
+        )
+
+        if self.chemistry == pybamm.parameter_sets.Ai2020 and self.is_experiment:
+            batch_study.solve(calc_esoh=False)
+        else:
+            batch_study.solve([0, t_end])
+
+        create_gif(batch_study)
+
+        self.comparison_dict.update(
+            {"varied_values": [], "params": parameter_values_for_comp}
+        )
+
+    def parameter_comparison(self):
+        """
+        Generates a comparison with a single model, by varying a single parameter
+        """
+        # generate a list of parameter values by varying a single parameter
+        labels = []
+        varied_values = []
+        param_list = []
+        # randomly select number of comparisons and vary a
+        # parameter value
+        diff_params = random.randint(2, 3)
+        for i in range(0, diff_params):
+
+            # generate parameter values
+            params = parameter_value_generator(
+                self.parameter_values.copy(), {self.param_to_vary: self.bounds}
+            )
+
+            # append the varied values in `labels` which will be used
+            # in the GIF
+            labels.append(self.param_to_vary + ": " + str(params[self.param_to_vary]))
+            varied_values.append(params[self.param_to_vary])
+
+            # create a list of ParameterValues with each element
+            # having the same parameter varied
+            param_list.append(params)
+
+        # convert the list containing parameter values to a dictionary
+        # for pybamm.BatchStudy
+        parameter_values_for_comp = dict(list(enumerate(param_list)))
+
+        # calculate t_end if varying "Current function [A]"
+        if self.param_to_vary == "Current function [A]":
+            t_end = self.calculate_t_end(parameter_values_for_comp)
+        else:
+            t_end = 3700
+
+        batch_study = pybamm.BatchStudy(
+            models=self.models_for_comp,
+            parameter_values=parameter_values_for_comp,
+            experiments=self.experiment,
+            permutations=True,
+        )
+
+        if self.chemistry == pybamm.parameter_sets.Ai2020 and self.is_experiment:
```

Comment:
here you could do
```python3
if self.is_experiment:
    if self.chemistry == pybamm.parameter_sets.Ai2020:
        batch_study.solve(calc_esoh=False)
    else:
        batch_study.solve()
else:
    t_end = self.calculate_t_end(parameter_values_for_comp)
    batch_study.solve([0, t_end])
``` 

then you could do 
```python3
    def calculate_t_end(self, parameter_values_for_comp):
        """
        Calculates the t_end for t_eval (t_eval=[0, t_end])
        Parameter:
            parameter_values_for_comp: dict
                Of the form -
                {
                    0: pybamm.ParameterValues,
                    1: pybamm.ParameterValues
                }
        """
        if self.param_to_vary == "Current function [A]"
            # find the minimum value for "Current function [A]"
            min_curr_value = min(
                [
                    item["Current function [A]"]
                    for k, item in parameter_values_for_comp.items()
                ]
            )
            factor = min_curr_value / self.parameter_values["Current function [A]"]
            t_end = (1 / factor * 1.1) * 3600
       else:
            t_end = 3700
        return t_end
```

Reply:
Thanks! The `self.param_to_vary == "Current function [A]"` condition might create an issue in `"model comparison"` (where `"Current function [A]"` and `"Ambient temperature [K]"` are always varied if not `is_experiment`) as `self.param_to_vary` remains `None` there. Maybe I can pass another parameter `force` and the condition can be like `if self.param_to_vary == "Current function [A]" or force:`?